internal/listtoken/item.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package listtoken
+
+import (
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/types/resource"
+)
+
+// Item defines a subset of a boundary.Resource that can
+// be used as an input to a DB operation for the purposes
+// of pagination and sorting.
+type Item interface {
+	GetPublicId() string
+	GetCreateTime() *timestamp.Timestamp
+	GetUpdateTime() *timestamp.Timestamp
+	GetResourceType() resource.Type
+}
```

Comment:
I had to move this from `pagination` to this package because we need it in this package and can't import the `pagination` package here. It doesn't seem to have caused any problems.

Reply:
The problem is that the `Transition` function needs to take an `Item` parameter. We can't take an unexported `item`, since it's used from the pagination package. We could take an `any` and then type assert to the unexported `item` but that seems worse.

Reply:
It looks like `Transition` only needs a `boundary.Resource` since it only calls `GetPublicId()` and `GetCreateTime()`.

Reply:
That works :+1:

Reply:
Done